openedx/core/djangoapps/user_api/accounts/api.py

```diff
@@ -843,31 +825,36 @@ def update_position_settings(request):
 
     data = request.data
     position = data.get('position')
+    sub_position = data.get('sub_position')
     other_position = data.get('other_position')
     specialization = data.get('specialization')
     diploma_id = data.get('diploma_id')
 
     user = UserProfile.objects.get(user=request.user)
 
+    user_position = user.position
+
     if position:
         position = Position.objects.get(id=position)
+
+        if sub_position and user_position is not None and position in [user_position, user_position.parent]:
```

Comment:
Can we use in the condition just 
```
if sub_position and user_position and position in [user_position, user_position.parent]
```
or just
```
if sub_position and position in [user_position, user_position.parent]
```

Reply:
Can't do it. user.position can be None. In this case, we will have an exception when trying to get a `parent` attribute from None.

Reply:
Ah, I see it.